src/biz/access_request/ops.rs

```diff
@@ -16,18 +16,56 @@ use shared_entity::dto::access_request_dto::{AccessRequest, AccessRequestView};
 use sqlx::PgPool;
 use uuid::Uuid;
 
-use crate::biz::collab::{
-  folder_view::{to_dto_view_icon, to_view_layout},
-  ops::get_latest_collab_folder,
+use crate::{
+  biz::collab::{
+    folder_view::{to_dto_view_icon, to_view_layout},
+    ops::get_latest_collab_folder,
+  },
+  mailer::{Mailer, WorkspaceAccessRequestMailerParam},
 };
 
 pub async fn create_access_request(
   pg_pool: &PgPool,
+  mailer: Mailer,
+  appflowy_web_url: &str,
   workspace_id: Uuid,
   view_id: Uuid,
   uid: i64,
 ) -> Result<Uuid, AppError> {
   let request_id = insert_new_access_request(pg_pool, workspace_id, view_id, uid).await?;
+  let access_request = select_access_request_by_request_id(pg_pool, request_id).await?;
+  let cloned_mailer = mailer.clone();
+  let approve_url = format!(
+    "{}/app/approve-request?request_id={}",
+    appflowy_web_url, request_id
+  );
+  let email = access_request.workspace.owner_email.clone();
+  let recipient_name = access_request.workspace.owner_name.clone();
+  // use default icon until we have workspace icon
+  let workspace_icon_url =
+    "https://miro.medium.com/v2/resize:fit:2400/1*mTPfm7CwU31-tLhtLNkyJw.png".to_string();
+  let user_icon_url =
+    "https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_1280.png"
+      .to_string();
```

Comment:
I think we can find the icons in the `metadata` column of `af_user` and `icon` column of `af_workspace`, fallback this if not found

Reply:
One problem with this approach, is that the url metadata currently being stored on the user profile is not actually an url, but emoji UTF-8 string. We will properly need proper image support before we can do this.